config/v1/types_infrastructure.go

```diff
@@ -1417,6 +1419,89 @@ type NutanixPlatformSpec struct {
 	// +listType=map
 	// +listMapKey=name
 	PrismElements []NutanixPrismElementEndpoint `json:"prismElements"`
+
+	// failureDomains configures failure domains information for the Nutanix platform.
+	// When set, the failure domains defined here may be used to spread Machines across
+	// prism element clusters to improve fault tolerance of the cluster.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	FailureDomains []NutanixFailureDomain `json:"failureDomains"`
+}
+
+// NutanixFailureDomain configures failure domain information for the Nutanix platform.
+type NutanixFailureDomain struct {
+	// name defines the unique name of a failure domain.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:Pattern=`[a-z0-9]([-a-z0-9]*[a-z0-9])?`
+	Name string `json:"name"`
+
+	// prismCentralEndpoint holds the endpoint (address, port) to access the Nutanix Prism Central.
+	// If this is not configured, it means this failure domain uses the default prism_central
+	// endpoint as configured in the nutanix platformSpec.
+	// +optional
+	PrismCentralEndpoint *NutanixPrismEndpoint `json:"prismCentralEndpoint,omitempty"`
+
+	// prismElementEndpoint holds the endpoint address and port to access the Nutanix Prism Element.
+	// When a cluster-wide proxy is installed, by default, this endpoint will be accessed via the proxy.
+	// Should you wish for communication with this endpoint not to be proxied, please add the endpoint to the
+	// proxy spec.noProxy list.
+	// +optional
```

Comment:
Is this definitely optional? What happens when it isn't specified?

Reply:
Yes, the PE endpoint is optional. The nutanix-mapi-controller does not use it.  The only use case I can think of is when user configures the CSI credential secret data, they can reference the PE endpoint information.

Reply:
So what uses this information today? Do we really need this field if the Machine API and other components you're adjusting for this feature aren't using the information?

Reply:
Dropped this for now.